Scripts/build-webdriveragent.js

```diff
@@ -1,34 +1,31 @@
 const path = require('path');
-const os = require('os');
 const { asyncify } = require('asyncbox');
-const { logger, fs, mkdirp, zip } = require('@appium/support');
+const { logger, fs, zip } = require('@appium/support');
 const { exec } = require('teen_process');
 const xcode = require('appium-xcode');
 
 const log = new logger.getLogger('WDABuild');
 const rootDir = path.resolve(__dirname, '..');
+const derivedDataPath = `${rootDir}/wdaBuild`;
+const wdaAppBundle = 'WebDriverAgentRunner-Runner.app';
+const appBundlePath = path.join(derivedDataPath, 'Build', 'Products', 'Debug-iphonesimulator');
 
 async function buildWebDriverAgent (xcodeVersion) {
+  log.info(`Deleting ${derivedDataPath} if exists`);
+  if (await fs.exists(derivedDataPath)) {
```

Comment:
maybe it would make more sense to perform `clean` with xcodebuild instead?

Reply:
ah, will take a look later